autonomi/src/client/utils.rs

```diff
@@ -35,16 +35,25 @@ impl Client {
     pub(crate) async fn fetch_from_data_map(&self, data_map: &DataMap) -> Result<Bytes, GetError> {
         let mut encrypted_chunks = vec![];
 
-        for info in data_map.infos() {
-            let chunk = self
-                .chunk_get(info.dst_hash)
-                .await
-                .inspect_err(|err| error!("Error fetching chunk {:?}: {err:?}", info.dst_hash))?;
-            let chunk = EncryptedChunk {
-                index: info.index,
-                content: chunk.value,
-            };
-            encrypted_chunks.push(chunk);
+        let mut stream = futures::stream::iter(data_map.infos().into_iter())
+            .map(|info| {
+                let dst_hash = info.dst_hash;
+                async move {
+                    self.chunk_get(dst_hash)
+                        .await
+                        .inspect_err(move |err| {
+                            error!("Error fetching chunk {:?}: {err:?}", dst_hash)
+                        })
+                        .map(|chunk| EncryptedChunk {
+                            index: info.index,
+                            content: chunk.value,
+                        })
+                }
+            })
+            .buffered(*CHUNK_DOWNLOAD_BATCH_SIZE);
+
+        while let Some(encrypted_chunk) = stream.next().await {
```

Comment:
this while loop will terminate on the first `failed to fetch` as well,
which will give incorrect input to the `decrypt`
there shall be some error out for retry or print out to give more explicit error info the user ?

Reply:
The return type is a result. We get a `None` if the iter runs out. We do short circuit on error below.
I've renamed it to `encrypted_chunk_result` to be more accurate.